monkey/infection_monkey/exploit/drupal.py

```diff
@@ -0,0 +1,161 @@
+"""
+Remote Code Execution on Drupal server - CVE-2019-6340
+Implementation is based on:
+    https://gist.github.com/leonjza/d0ab053be9b06fa020b66f00358e3d88/f9f6a5bb6605745e292bee3a4079f261d891738a.
+"""
+
+import logging
+import requests
+from urllib.parse import urljoin
+from infection_monkey.exploit.web_rce import WebRCE
+from network.network_utils import remove_port
+
+__author__ = 'Ophir Harpaz'
+
+LOG = logging.getLogger(__name__)
+
+
+def check_drupal_cache(r: requests.Response) -> bool:
+    """
+    Check if a response had the cache header.
+    """
+    return 'X-Drupal-Cache' in r.headers and r.headers['X-Drupal-Cache'] == 'HIT'
+
+
+def find_articles(base_url: str, lower: int = 1, upper: int = 10):
+    """ Find a target article that does not 404 and is not cached """
```

Comment:
This comment is misleading (singular instead of plural)

Reply:
f31186272f42252efe0d4c0c23758b834f054179